src/containers/dashboard/sagas/getStaffCasesData.js

```diff
@@ -0,0 +1,221 @@
+      .filter((neighbor :Map) => getNeighborESID(neighbor) === formESID)
+      .map((neighbor :Map) => {
+        const form :Map = getNeighborDetails(neighbor);
+        if (getPropertyValue(form, [NAME, 0]) === REPAIR_HARM_AGREEMENT) {
+          return form;
+        }
+        return Map();
+      }));
+
+    const staffCasesTableData = List().withMutations((mutator) => {
+
+      let totalPendingIntake :number = 0;
+      let totalPendingCircle :number = 0;
+      let totalRepairHarmAgreementsCompleted :number = 0;
+      let totalOpenCases :number = 0;
+      let totalClosedCases :number = 0;
+
+      staff.forEach((staffMember :Map) => {
+        const staffEKID :?UUID = getEntityKeyId(staffMember);
+        const personName :string = getPersonName(staffMember);
+        const crcCases :List = crcCasesByStaffEKID.get(staffEKID, List());
+
+        let pendingIntake :number = 0;
+        let pendingCircle :number = 0;
+        let repairHarmAgreementsCompleted :number = 0;
+        let openCases :number = 0;
+        let closedCases :number = 0;
+
+        crcCases.forEach((crcCase :Map) => {
+          const crcCaseEKID :?UUID = getEntityKeyId(crcCase);
+          const statuses :List = statusesByCRCEKID.get(crcCaseEKID, List());
+          const intake :Map = statuses
+            .find((status) => getPropertyValue(status, [PropertyTypes.STATUS, 0]) === INTAKE);
+          const acceptance :Map = statuses
+            .find((status) => getPropertyValue(status, [PropertyTypes.STATUS, 0]) === ACCEPTANCE);
+          const circle :Map = statuses
+            .find((status) => getPropertyValue(status, [PropertyTypes.STATUS, 0]) === CIRCLE);
+          const closed :Map = statuses
+            .find((status) => getPropertyValue(status, [PropertyTypes.STATUS, 0]) === CLOSED);
+          const resolution :Map = statuses
+            .find((status) => getPropertyValue(status, [PropertyTypes.STATUS, 0]) === RESOLUTION);
+
+          if (isDefined(intake) && !isDefined(acceptance)) pendingIntake += 1;
+          if (isDefined(acceptance) && !isDefined(circle)) pendingCircle += 1;
+          if (!isDefined(closed) && !isDefined(resolution)) openCases += 1;
+          if (isDefined(closed) || isDefined(resolution)) closedCases += 1;
+
+          const repairHarmAgreement = repairHarmAgreementByCRCEKID.get(crcCaseEKID, Map());
+          if (isDefined(repairHarmAgreement) && !repairHarmAgreement.isEmpty()) {
+            repairHarmAgreementsCompleted += 1;
+          }
+        });
+
+        const tableRow = Map({
+          [STAFF_CASES_TABLE_HEADERS[0]]: personName,
+          [STAFF_CASES_TABLE_HEADERS[1]]: pendingIntake,
+          [STAFF_CASES_TABLE_HEADERS[2]]: pendingCircle,
+          [STAFF_CASES_TABLE_HEADERS[3]]: repairHarmAgreementsCompleted,
+          [STAFF_CASES_TABLE_HEADERS[4]]: openCases,
+          [STAFF_CASES_TABLE_HEADERS[5]]: closedCases,
```

Comment:
Super picky here, and no change required. I think this `STAFF_CASES_TABLE_HEADERS` constant should have been defined as an object and not an array. Retrieving the keys for the map by indexing into an array seems error since it assumes a specific ordering of the keys.

I think this is easier to comprehend.
```
const tableRow = Map({
   STAFF_CASES_TABLE_HEADERS.CLOSED_CASES: ...
   .....
})
```

or destructure and ...

```
const tableRow = Map({
   CLOSED_CASES: ...
   .....
})
```

Reply:
this does make more sense. i went ahead and made these changes!